### DISCUSSION TITLE
positionAndNormalFromPoint always returns `null`

### DISCUSSION THREAD

**Discussion by TalRodin** [google/model-viewer#2896] [Q&A]
Hi, could you please help me with panning? After running the positionAndNormalFromPoint function returns null always or throw an error. I saw online they advised to use layerX isntead of clientX. but at this point it does not return error but gives null and constantly return object to the position auto
```   const rect = modelViewer.getBoundingClientRect();
            const x = pointer.layerX - rect.left;
            const y = pointer.layerY- rect.top;
            console.log(x, y);
            const hit = modelViewer.positionAndNormalFromPoint(x, y);
            console.log(hit);
            modelViewer.cameraTarget =
                hit == null ? "auto auto auto" : hit.position.toString();```
7.125 196
ModelViewer.js:382 null
ModelViewer.js:380 7.125 196
ModelViewer.js:382 null
ModelViewer.js:380 7.125 196
ModelViewer.js:382 null

**Comment by samaneh-kazemi** [comment]
Hi @TalRodin, 

Based on [MDN docs](https://developer.mozilla.org/en-US/docs/Web/API/UIEvent/layerX) using layerX is not recommended. 

Can you try our using ClientX/Y instead? You can see our example [here](https://modelviewer.dev/examples/stagingandcameras/#panning)? Let us know if that doesn't fix this for you.

**Reply by TalRodin** [reply]
Hi i work on reacreating it. But with clientX/Y gives me errors in` Cannot read property 'matrixWorld' of undefined`
![image](https://user-images.githubusercontent.com/32145723/138143992-79341bf6-68c4-4141-b315-1ae943226384.png)

**Reply by TalRodin** [reply]
with layerX it does not give me errors, and sometimes returns values

**Reply by TalRodin** [reply]
<img width="150" alt="Screen Shot 2021-10-20 at 1 44 26 PM" src="https://user-images.githubusercontent.com/32145723/138144256-df8ad468-94c1-476a-adcc-8836f01b921d.png">

**Reply by TalRodin** [reply]
I cannot share link with you. As we use it for the project. I can try to recreate it.

**Reply by samaneh-kazemi** [reply]
@TalRodin it's hard to tell what is the issue without having a repro link. I'd recommend you validate your gltf file using our [editor](https://modelviewer.dev/editor/) as well to make sure that's not the issue.

**Reply by TalRodin** [reply]
@asemanh yes recreate the error for you in sandbox . i tried to download .glb into editor but it gave me erros. do those errors effect on smt
<img width="1010" alt="Screen Shot 2021-10-20 at 2 23 55 PM" src="https://user-images.githubusercontent.com/32145723/138149958-488d655f-acff-4712-b56d-f876d230c510.png">
h?

**Reply by samaneh-kazemi** [reply]
@TalRodin can you try a simple gltf and make sure editor works for you? (you can simply download one from our editor). This way we can confirm that the issue is not on the editor.
